### PR TITLE
Issue 1194 repeater frozen column

### PR DESCRIPTION
**Pull request by BenjaminNeilDavis**
Creation of new frozen plugin for repeater

### CODE REVIEW

File: js/repeater-frozen.js [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+			frozen_noItemsHTML: 'no items found',
+			frozen_selectable: false,
+			frozen_sortClearing: false,
+			frozen_rowRendered: null,
+			frozen_frozenColumns: 0
+		});
+
+		//EXTENSION DEFINITION
+		$.fn.repeater.viewTypes.frozen = {
+			cleared: function () {
```
Comment (kevinparkerson) [inline js/repeater-frozen.js]: This should be removed altogether unless something will reside within the function

File: js/repeater-frozen.js [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+					options.sortProperty = this.frozen_sortProperty;
+				}
+				return options;
+			},
+			initialize: function (helpers, callback) {
+				this.frozen_sortDirection = null;
+				this.frozen_sortProperty = null;
+				callback();
+			},
+			resize: function () {
```
Comment (kevinparkerson) [inline js/repeater-frozen.js]: This should be removed altogether unless something will reside within the function

**Comment by kevinparkerson** [comment]
Looking pretty good! I noticed a few minor issues.

1) In the dev.html page, the repeater's footer content was extending beyond the footer itself. This appeared to resolve itself when resizing the browser, so it's probably a fluke of the repeater's sizing method, which can be solved by calling "resize" at the appropriate time. Doubt it's due to the plugin, but wanted to make note of it.

2) In Firefox, the frozen column header is missing a right border. There is also a little wonkiness with the frozen column shifting around a bit.

File: js/repeater-list.js [hunk 3]
```diff
@@ -202,6 +254,11 @@
 							self.list_positionHeadings();
 						}
 					});
+					if (self.viewOptions.list_frozenColumns) {
+						helpers.container.on('scroll.fu.repeaterFrozen', function () {
```
Comment (kevinparkerson) [inline js/repeater-list.js]: This event should be `scroll.fu.repeaterList`. Probably a symptom of moving code from the repeater-frozen file into the repeater-list plugin

**Comment by kevinparkerson** [comment]
Commented on the event naming, outside that looks good!

**Comment by kevinparkerson** [comment]
Update: we might have to consider row selection as well here

**Comment by BenjaminNeilDavis** [comment]
To use the frozen columns:
- Use the list view option `list_frozenColumns: n ` where n is the number of columns to freeze. These must be the columns on the farthest left.
- The frozen columns table is set to table-layout: fixed; which helps with the sizing of the frozen column and to keep from changing the sizes of the columns between page switches. With this enabled you can set the width of the columns with something like this:
```
table thead tr th, .repeater-list-heading {
width: 200px;
}
```

**Pull request merged**